jenkins/version-def.sh

```diff
@@ -31,6 +31,11 @@ CUDA_CLASSIFIER=${CUDA_CLASSIFIER:-"cuda11"}
 PROJECT_VER=${PROJECT_VER:-"21.10.0-SNAPSHOT"}
 PROJECT_TEST_VER=${PROJECT_TEST_VER:-"21.10.0-SNAPSHOT"}
 SPARK_VER=${SPARK_VER:-"3.0.1"}
+# Make a best attempt to set the default value for the shuffle shim.
+# Note that SPARK_VER for non-Apache Spark flavors (i.e. databricks,
+# cloudera, and others) may not be a simple as just the version number, so
+# this variable should be set accordingly.
+SHUFFLE_SPARK_SHIM=${SHUFFLE_SPARK_SHIM:-spark${SPARK_VER//./}}
```

Comment:
no csp version as default is reasonable. can we add default to cover spark snapshot versions here? maybe something like 

`${SHUFFLE_SPARK_SHIM:-spark${SPARK_VER//[.|\-SNAPSHOT]/}}`

Reply:
I am not an expert at bash pattern matching, but searching around yielded this:
`${SHUFFLE_SPARK_SHIM:-spark${SPARK_VER//?(.|\-SNAPSHOT)/}}`

the pattern you provided matches the letters S,N,A,P,S,H,O,T in any order, the pattern above specifically looks for `-SNAPSHOT` and trims it in as one unit.